pkg/sql/descriptor.go

```diff
@@ -143,23 +141,43 @@ func (p *planner) createDescriptorWithID(
 		return err
 	}
 
-	if mutType, ok := descriptor.(*sqlbase.MutableTypeDescriptor); ok {
-		if err := mutType.Validate(ctx, p.txn, p.ExecCfg().Codec); err != nil {
+	mutDesc, ok := descriptor.(catalog.MutableDescriptor)
+	if !ok {
+		log.Fatalf(ctx, "unexpected type %T when creating descriptor", descriptor)
+	}
+	isTable := false
+	switch desc := mutDesc.(type) {
+	case *sqlbase.MutableTypeDescriptor:
+		if err := desc.Validate(ctx, p.txn, p.ExecCfg().Codec); err != nil {
 			return err
 		}
-		if err := p.Descriptors().AddUncommittedDescriptor(mutType); err != nil {
+		if err := p.Descriptors().AddUncommittedDescriptor(mutDesc); err != nil {
 			return err
 		}
-	}
-
-	mutDesc, isTable := descriptor.(*sqlbase.MutableTableDescriptor)
-	if isTable {
-		if err := mutDesc.ValidateTable(); err != nil {
+	case *sqlbase.MutableTableDescriptor:
+		isTable = true
+		if err := desc.ValidateTable(); err != nil {
 			return err
 		}
 		if err := p.Descriptors().AddUncommittedDescriptor(mutDesc); err != nil {
 			return err
 		}
+	case *sqlbase.MutableDatabaseDescriptor:
+		if err := desc.Validate(); err != nil {
+			return err
+		}
+		if p.Descriptors().DatabaseLeasingUnsupported() {
+			p.Descriptors().AddUncommittedDatabaseDeprecated(desc.Name, desc.ID, descs.DBCreated)
+		} else {
+			if err := p.Descriptors().AddUncommittedDescriptor(mutDesc); err != nil {
+				return err
+			}
+		}
+	case *sqlbase.MutableSchemaDescriptor:
+		// TODO (lucy): Integrate with schema leasing. At that point, we can pull
+		//  the AddUncommittedDescriptor calls out of the individual cases above.
```

Comment:
we still need to call adduncommitted descriptor here for schemas (looks like a mistake on my part to not include it here)